packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.js

```diff
@@ -252,11 +275,21 @@ class CryptoWatchPriceFeed extends PriceFeedInterface {
         // Sorts the data such that the oldest elements come first.
         return a.openTime - b.openTime;
       });
+  }
 
-    // 5. Store results.
-    this.currentPrice = newPrice;
-    this.historicalPricePeriods = newHistoricalPricePeriods;
-    this.lastUpdateTime = currentTime;
+  _computeTwap(endTime, ohlcs) {
+    // Combine open and close to get more data fidelity at the edges of the range.
+    const priceTimes = lodash.flatten(
+      ohlcs.map(pricePeriod => {
+        return [
+          [pricePeriod.openTime, pricePeriod.openPrice],
+          [pricePeriod.closeTime, pricePeriod.closePrice]
```

Comment:
Is there any time between a period's `closeTime` and the next's `openTime`?

Reply:
There shouldn't be unless:
- An ohlc point is missing.
- The data point is the first or last, so there is only an `openTime` (beginning of the data) or a `closeTime` (end of the data).

Both of these shouldn't affect the output too much, but I think including both makes the calculation a little more robust.

Reply:
just as a side note, i find it very weird that historical data is represented as ohlc if closeTime/ next openTime are identical and open/close price are identical. its really redundant. I'm assuming that its this way because they want the ability to differentiate between close/open price/time if needed.

Reply:
Yeah, I'm really not sure why it's represented this way. This is the traditional "candle" way of viewing data. My guess is that it mostly has to do with the fact that they want a single candle to be self-contained. It's also somewhat unclear what they mean by price -- is it trade price or midpoint. If it's midpoint, it's pretty much redundant. If it's first/last trade, then I could see them being different values and having some value in being distinguished.